Response-Theory/Self-Consistent-Field/helper_CPHF.py

```diff
@@ -273,97 +270,110 @@ def solve_static_iterative(self, maxiter=20, conv=1.e-9, use_diis=True):
             print('CPHF Iteration %3d: Average RMS = %3.8f  Maximum RMS = %3.8f' %
                   (CPHF_ITER, avg_RMS, max_RMS))
 
-    def solve_dynamic_iterative(self, omega=0.0, maxiter=20, conv=1.e-9, use_diis=True):
+    def solve_dynamic_iterative(self, omega=0.0, maxiter=20, conv=1.e-10, use_diis=True):
 
         # Init JK object
         jk = psi4.core.JK.build(self.scf_wfn.basisset())
         jk.initialize()
 
-        # Add blank matrices to the JK object and NumPy hooks to
-        # C_right; there are 6 sets of matrices to account for X and Y
-        # vectors separately.
-        npC_right = []
-        for xyz in range(6):
-            jk.C_left_add(self.Co)
-            mC = psi4.core.Matrix(self.nbf, self.nocc)
-            npC_right.append(np.asarray(mC))
-            jk.C_right_add(mC)
-
-        # Build initial guess, previous vectors, diis object, and C_left updates
-        x_l, x_r = [], []
-        x_l_old, x_r_old = [], []
-        diis_l, diis_r = [], []
-        ia_denom_l = self.epsilon[self.nocc:] - self.epsilon[:self.nocc].reshape(-1, 1) - omega
-        ia_denom_r = self.epsilon[self.nocc:] - self.epsilon[:self.nocc].reshape(-1, 1) + omega
-        for xyz in range(3):
-            x_l.append(self.dipoles_xyz[xyz] / ia_denom_l)
-            x_r.append(self.dipoles_xyz[xyz] / ia_denom_r)
-            x_l_old.append(np.zeros(ia_denom_l.shape))
-            x_r_old.append(np.zeros(ia_denom_r.shape))
-            diis_l.append(DIIS_helper())
-            diis_r.append(DIIS_helper())
+        # Build initial guess, previous vectors, and DIIS objects
+        norb = self.scf_wfn.nmo()
+        C = np.asarray(self.C)
+        rhsmats = [C.T @ np.asarray(dipmat) @ C for dipmat in self.tmp_dipoles]
```

Comment:
Could you use `np.dot` here. I do like the `@` syntax better, just not sure if people are ready for it yet.